README.md

```diff
@@ -76,6 +76,14 @@ Tenant ID: diku
 Username: diku
 ```
 
+#### API Key Sources
+
+The API Key can be specified as either:
+
+1) The `apiKey` query argument, e.g. `https://.../validate?apiKey=QlBhb2ZORm5jSzY0NzdEdWJ4RGhfZGlrdV9kaWt1`
+1) The `Authorization` request header, e.g. `Authorization: QlBhb2ZORm5jSzY0NzdEdWJ4RGhfZGlrdV9kaWt1`
```

Comment:
Should we follow the convention of "Authorization: <type> <credentials>" where **type** := **Basic** | **Bearer** | **Digest** among others, and **credentials** is the API Key?  Basic makes no attempt to encrypt the credentials, but the key itself can be base64-encoded, which is what we have.  [](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization )

Reply:
Bearer makes sense to me.

Reply:
Should we dictate how they call our service because we're the ones creating it? :)  I think we need to check with them to see how they use headers. If they don't want to modify the code then maybe we have to conform to what they are doing? But that's not a best practice.

Reply:
There's no way we're ever going to meet ever external system's needs all the time.  The best we can do is provide out-of-the box support for a standard approach to passing API keys via request headers.  Unfortunately, there doesn't seem to be consensus from the internet as to what the best way is of doing this, so we have to just pick one that makes sense to us.  

If, an Edge API needs is being used to integrate with an outside system, and that system can't provide the API key via one of these ways, the Edge API developer can simply override the `ApiKeyHelper` class accordingly.  Since the ApiKeyHelper is passed into the handler(s), the developer has the opportunity to provide their own implementation.  I just pushed a few changes that should make this sort of thing even easier...

The `getFromParam(RoutingContext)`, `getFromHeader(RoutingContext)`, and `getFromPath(RoutingContext)` methods could simply be overridden with custom logic, e.g. look in the X-Api-Key, or expect some authorization type specifier (apiKey, bearer, etc.) and return just the apiKey itself.

I also changed the behavior for `getFromHeader(RoutingContext)` to look for the "apikey" authorization type specifier, and return just the api key if present.  If the apikey type specifier isn't provided, the entire header value is returned...  So both:

`Authorization: apikey abcd12345`
and
`Authorization: abcd12345`

will work.  In both cases, abc12345 would be returned.  The type check is case insensitive, so APIKEY, apikey, ApiKey, apiKey would all work.